packages/ozone/src/api/set/upsertSet.ts

```diff
@@ -0,0 +1,44 @@
+import { AuthRequiredError, InvalidRequestError } from '@atproto/xrpc-server'
+import { Server } from '../../lexicon'
+import AppContext from '../../context'
+
+export default function (server: Server, ctx: AppContext) {
+  server.tools.ozone.sets.upsertSet({
+    auth: ctx.authVerifier.modOrAdminToken,
+    handler: async ({ input, auth }) => {
+      const access = auth.credentials
+      const db = ctx.db
+      const { name, description = '' } = input.body
+
+      if (!access.isModerator) {
+        throw new AuthRequiredError(
+          'Must be a moderator to create or update a set',
+        )
+      }
+
+      if (!name) {
+        throw new InvalidRequestError('Name is required')
+      }
+
+      const setService = ctx.setService(db)
+
+      const upsertedSet = await db.transaction(async (txn) => {
+        return await setService.upsert(
+          {
+            name,
+            description,
+          },
+          txn,
+        )
+      })
+
+      return {
+        encoding: 'application/json',
+        body: {
+          name: upsertedSet.name,
+          description: upsertedSet.description ?? undefined,
+        },
```

Comment:
Since `description` defaults to `''` above, I wonder if this could end-up being an empty string, which isn't allowed by the lexicon.